consumer.py

```diff
@@ -1,97 +1,36 @@
-import pika
-import io
+from async_consumer import AsyncConsumer
 import logging
-import sys
+from structlog import wrap_logger
 import settings
-import requests
-import zipfile
-from ftplib import FTP
+from response_processor import ResponseProcessor
 
-logging.basicConfig(stream=sys.stdout, level=settings.LOGGING_LEVEL, format=settings.LOGGING_FORMAT)
+logging.basicConfig(level=settings.LOGGING_LEVEL, format=settings.LOGGING_FORMAT)
 
-logging.debug("sdx-downstream|START")
+logger = logging.getLogger(__name__)
 
-def transform_xml(stored_json):
-    transform_url = "%s/xml" % (settings.SDX_TRANSFORM_TESTFORM_URL)
-    transformed_data = requests.post(transform_url, json=stored_json)
-    queue_notification(transformed_data.content)
+logger = wrap_logger(logger)
+logger.debug("START")
 
-def queue_notification(notification):
-    connection = pika.BlockingConnection(pika.URLParameters(settings.RABBIT_URL))
-    channel = connection.channel()
-    channel.queue_declare(queue=settings.RABBIT_QUEUE_TESTFORM)
-    channel.basic_publish(exchange='',
-                          properties=pika.BasicProperties(content_type='application/xml'),
-                          routing_key=settings.RABBIT_QUEUE_TESTFORM,
-                          body=notification)
-    logging.debug(notification)
-    connection.close()
 
-def transform_cs(stored_json, sequence_no):
-    transform_url = "%s/common-software/%d" % (settings.SDX_TRANSFORM_CS_URL, sequence_no)
-    transformed_data = requests.post(transform_url, json=stored_json)
-    zip_contents = transformed_data.content
-
-    try:
-        z = zipfile.ZipFile(io.BytesIO(zip_contents))
-        logging.debug("Zip contents:")
-        logging.debug(z.namelist())
-        ftp = connect_to_ftp()
-        for filename in z.namelist():
-            if filename.endswith('/'):
-                continue
-            logging.debug("Processing file from zip: " + filename)
-            edc_file = z.open(filename)
-            deliver_binary_to_ftp(ftp, filename, edc_file.read())
-        ftp.quit()
-
-    except (RuntimeError, zipfile.BadZipfile):
-        logging.debug("Bad zip file!")
-            # TODO: Need to deal with exception
-
-def connect_to_ftp():
-    ftp = FTP(settings.FTP_HOST)
-    ftp.login(user=settings.FTP_USER, passwd=settings.FTP_PASS)
-    return ftp
-
-
-def deliver_ascii_to_ftp(ftp, filename, data):
-    stream = io.StringIO(data)
-    ftp.storlines('STOR ' + filename, stream)
+class Consumer(AsyncConsumer):
+    def on_message(self, unused_channel, basic_deliver, properties, body):
 
+        logger.info('Received message # %s from %s: %s' % (basic_deliver.delivery_tag, properties.app_id, body))
 
-def deliver_binary_to_ftp(ftp, filename, data):
-    stream = io.BytesIO(data)
-    ftp.storbinary('STOR ' + filename, stream)
+        manager = ResponseProcessor(logger)
+        proccessed_ok = manager.process(body.decode("utf-8"))
```

Comment:
Why `manager`? I think either `ResponseProcessor` should be `ResponseManager`, or the variable should be `processor`